app/src/main/java/com/epfl/drawyourpath/database/FireDatabase.kt

```diff
@@ -214,6 +215,41 @@ class FireDatabase : Database() {
         return updateUserData(dataUpdated)
     }
 
+    override fun addUserToFriendsList(userId: String): CompletableFuture<Unit> {
+        val future = CompletableFuture<Unit>()
+        return isUserStoredInDatabase(userId).thenApply {
+            if(!it){
+                throw Exception("The user with $userId is not present on the database.")
+            }else{
+                //add the user to the the friendList of the current user
+                val currentUserId = getUserId()
+                if(currentUserId==null){
+                    throw Exception("The userId can't be null !")
+                }else{
+                    addUserIdToFriendList(currentUserId, userId).thenApply {
+                        //add the currentUser to the friend list of the user with userId
+                        addUserIdToFriendList(userId, currentUserId)
+                    }
+                }
+            }
+        }
+    }
+
+    override fun removeUserFromFriendlist(userId: String): CompletableFuture<Unit> {
+        val currentUserId = getUserId()
+        if(currentUserId==null){
+            val future = CompletableFuture<Unit>()
+            future.completeExceptionally(Exception("The userId can't be null !"))
```

Comment:
I think the error message is misleading. It isn't the `userId` parameter that is `null`, but the current user's which means the user isn't logged.